### PR TITLE
[Feature] Performing uPort login and request flows 

### PR DESCRIPTION
**Pull request by ugoamanoh**
### What's Here
Fixes https://www.pivotaltracker.com/story/show/164276970

Implements the `Transports.sendExpectingResult()` in making uPort requests. The request flows are initiated by a SelectiveDisclosureRequest (uPort Login), after which the user can perform other request types such as `Verified Claim`,`Personal Signature`,`Typed Data` and, `Ethereum Transaction`

### CODE REVIEW

File: demoapp/src/main/java/me/uport/sdk/demoapp/request_flows/EthrTransactionUtils.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * I returns a Network and Address `Pair`
+ * and throws [IllegalArgumentException]
+ */
+fun getNetworkAndAddressFromDID(did: String): Pair<String, String> {
+
+    // converts possible ethr DIDs to a Pair of Network and Address
+    val ethrMatchResult = EthrDIDResolver.identityExtractPattern.find(did)
+    if (ethrMatchResult != null) {
+        val (address,_) = ethrMatchResult.destructured
+        return (Networks.rinkeby.networkId to address)
```
[the remark this answers is not in the document]
Comment (ugoamanoh) [reply demoapp/src/main/java/me/uport/sdk/demoapp/request_flows/EthrTransactionUtils.kt]: Would it be possible to get the network in the Selective Disclosure response?

[the remark this answers is not in the document]
Comment (mirceanis) [reply demoapp/src/main/java/me/uport/sdk/demoapp/request_flows/EthrTransactionUtils.kt]: I think it's the other way around.
You can request a particular network in a selective disclosure and the uPort app would create n account on that network

[the remark this answers is not in the document]
Comment (ugoamanoh) [reply demoapp/src/main/java/me/uport/sdk/demoapp/request_flows/EthrTransactionUtils.kt]: ok

**Comment by ugoamanoh** [comment]
@mirceanis All done

**Pull request merged**